torch/utils/_sympy/value_ranges.py

```diff
@@ -860,7 +860,28 @@ def bound_sympy(
     if context and context.fake_mode.shape_env:
         ranges = {**context.fake_mode.shape_env.var_to_range, **ranges}
 
-    unbounded_vars = expr.free_symbols - ranges.keys()
+    # sympy.Expr equality is done via pre-order traversal and hash of each item, which loads cpu => switching to str .name comparison with optimization for len(free_symbols) == 1
+    # unbounded_vars = expr.free_symbols - ranges.keys()
+    unbounded_vars = set()
+    num_fs = len(expr.free_symbols)
+    if  num_fs == 0:
+        pass
+    elif num_fs == 1:
+        fs = next(iter(expr.free_symbols))
+        name = fs.name
+        found = False
+        for r in ranges.keys():
+            if r.name == name:
+                found = True
+                break
```

Comment:
It's so nuts that this is faster. cc @lezcano

Reply:
lol cc @asmeurer @isuruf 

Also, `if not any(...): add`

Reply:
No, a symbol is identified by its name in dynamo/inductor.

Reply:
As Isuru noted, Symbols are compared by name + assumptions. That comparison should be fast, but I suppose it's possible there is some performance issue somewhere. It would be useful to have more precise timing information here.